### PR TITLE
docs: link radio driver docs

### PR DESCRIPTION
**Pull request by greg-fer**
Add links to new RD docs in nrfxlib and Infocenter.
Reworks updates from PR#2299 (KRKNWK-5824) due to TECHDOC-1544.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ug_thread.rst [hunk 1]
```diff
@@ -379,39 +340,12 @@ Antenna diversity
...
-    Each of them can be individually disabled or enabled, and the antenna manually selected.
-    Additionally, RX operations support automatic mode, where optimal antenna is selected for each individual operation.
+For more information, read the `antenna diversity documentation`_.
 
 Configuring antenna diversity
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The antenna diversity module is enabled by default for TX operations.
-
+In |NCS|, the antenna diversity module is enabled by default for TX operations.
```
Comment (MarekPorwisz) [inline doc/nrf/ug_thread.rst:348]: Maybe we could change it to something like "The antenna for TX operations needs to be controlled manually by the application". Current sentence does not imply there is no auto mode for TX

File: doc/nrf/ug_thread.rst [hunk 2]
```diff
@@ -58,10 +58,10 @@ Required modules
 Thread requires the following Zephyr's modules to properly operate in |NCS|:
 
 * :ref:`zephyr:ieee802154_interface` radio driver - This library is automatically enabled when working with OpenThread on Nordic Semiconductor's Development Kits.
+  Its advanced features are implemented through :ref:`nrfxlib:nrf_802154_sl` in `nrfxlib`_.
```
Comment (ru-fu) [inline doc/nrf/ug_thread.rst:61]:
```suggestion
  Its advanced features are implemented through the :ref:`nrfxlib:nrf_802154_sl` in `nrfxlib`_.
```
It resolves to "... Service Layer", correct?

File: doc/nrf/ug_thread.rst [hunk 3]
```diff
@@ -240,41 +240,12 @@ The radios of the device can use different modulation types, protocols, and over
...
-  For receive operations, and when there is no 802.15.4 radio activity, this pin is driven high.
-  When the coexistence interface is disabled, this pin is driven high.
-  Type of the output and the pin number to be used is configurable at compile time.
-* GRANT pin - This is the input of the 802.15.4 device and the output of the PTA.
-  This pin is activated by the PTA when it grants access to the radio medium to the 802.15.4 device.
-  The active state of the pin is fixed to low.
-  The pin number to be used is configurable at compile time.
-
-For more details about Wi-Fi coexistence, read the radio driver documentation.
+For more information, read `Wi-Fi coexistence documentation`_.
```
Comment (ru-fu) [inline doc/nrf/ug_thread.rst:243]:
```suggestion
For more information, read the `Wi-Fi coexistence documentation`_.
```

File: doc/nrf/ug_thread.rst [hunk 4]
```diff
@@ -300,25 +271,15 @@ To select a different behavior, use the following options:
...
-The third signal disables the FEM during the radio inactivity, in order to save energy.
-
-When the FEM is enabled, the radio driver toggles the three GPIO pins, based on the radio operation.
-The configuration structure allows to choose the pin polarity, and time differences between the pin toggle and the intended radio operation.
-
-.. figure:: images/pa-lna-block-dia.svg
-   :alt: Interfacing PA/LNA with nRF52
-
-   Interfacing PA/LNA with nRF52
+For more information, read `PA/LNA module documentation`_.
```
Comment (ru-fu) [inline doc/nrf/ug_thread.rst:277]:
```suggestion
For more information, read the `PA/LNA module documentation`_.
```

File: doc/nrf/ug_thread.rst [hunk 5]
```diff
@@ -379,39 +340,12 @@ Antenna diversity
...
-    Each of them can be individually disabled or enabled, and the antenna manually selected.
-    Additionally, RX operations support automatic mode, where optimal antenna is selected for each individual operation.
+For more information, read the `antenna diversity documentation`_.
 
 Configuring antenna diversity
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The antenna diversity module is enabled by default for TX operations.
-
+In |NCS|, the antenna diversity TX operations need to be controlled manually by the application.
```
Comment (ru-fu) [inline doc/nrf/ug_thread.rst:348]:
```suggestion
In |NCS|, the antenna diversity TX operations must be controlled manually by the application.
```

File: doc/nrf/ug_zigbee.rst [hunk 6]
```diff
@@ -41,12 +41,16 @@ See :ref:`nrfxlib:zboss` for more information.
...
-|zboss_lib|
-This library is available in `nrfxlib`_.
+Zigbee requires the following modules to properly operate in |NCS|:
 
-The OSIF subsystem acts as the linking layer between the ZBOSS stack and |NCS|.
-It implements a series of functions used by ZBOSS and is included in the |NCS|'s Zigbee subsystem.
-The files that handle the OSIF integration are located in :file:`subsys/zigbee/osif`.
+* :ref:`nrfxlib:zboss` available in nrfxlib, with the OSIF subsystem acting as the linking layer between the ZBOSS stack and |NCS|.
+  OSIF implements a series of functions used by ZBOSS and is included in the |NCS|'s Zigbee subsystem.
+  The files that handle the OSIF integration are located in :file:`subsys/zigbee/osif`.
```
Comment (ru-fu) [inline doc/nrf/ug_zigbee.rst:48]:
```suggestion
  The files that handle the OSIF integration are located in :file:`nrf/subsys/zigbee/osif`.
```
To make it clearer that this is in the nrf repo and not nrfxlib?

File: doc/nrf/ug_thread.rst (hunk 2 above)
Comment (greg-fer) [reply doc/nrf/ug_thread.rst:61]: Correct.

File: doc/nrf/ug_zigbee.rst (hunk 6 above)
Comment (greg-fer) [reply doc/nrf/ug_zigbee.rst:48]: Good point, I was wondering how to make it more clear, thanks.

**Comment by mbolivar-nordic** [comment]
Overriding CI as a docs only change

**Pull request merged**